SwiftWisdom/Core/StandardLibrary/Array/Array+Utilities.swift

```diff
@@ -22,55 +28,74 @@ public extension Array {
 }
 
 public extension Array where Element: Equatable {
-    public mutating func ip_remove(_ objectToRemove: Element) -> Bool {
-        for (idx, objectToCompare) in enumerated() where objectToRemove == objectToCompare {
+    /// Removes a single element from the array.
+    ///
+    /// - parameter object: Element to remove
+    ///
+    /// - returns: Boolean value indicating the success/failure of removing the element.
+    public mutating func ip_remove(object: Element) -> Bool {
+        for (idx, objectToCompare) in enumerated() where object == objectToCompare {
             remove(at: idx)
             return true
         }
         return false
     }
-    
-    public mutating func ip_removeElements(elements: [Element]) {
+
+    /// Removes multiple elements from an array.
+    ///
+    /// - parameter elements: Array of elements to remove
+    public mutating func ip_remove(elements: [Element]) {
         self = self.filter { element in
             return !elements.contains(element)
         }
     }
-    
-    /**
-     Returns NSNotFound for any element in elements that does not exist.
-     
-     - parameter elements: Array of Equatable elements
-     
-     - returns: Array of indexes or NSNotFound if element does not exist in self; count is equal to the count of `elements`
-     */
-    public func ip_indicesOf(elements: [Element]) -> [Int] {
+
+    /// Returns NSNotFound for any element in elements that does not exist.
+    ///
+    /// - parameter elements: Array of Equatable elements
+    ///
+    /// - returns: Array of indexes or NSNotFound if element does not exist in self; count is equal to the count of `elements`
+    public func ip_indices(ofElements elements: [Element]) -> [Int] {
         return elements.map { element in
-                return index(of: element) ?? NSNotFound
-            }
+            return index(of: element) ?? NSNotFound
+        }
     }
 }
 
 public extension Array where Element: Hashable {
+    /// Converts an Array into a Set of the same type.
+    ///
+    /// - returns: Set of the array's elements
     public func ip_toSet() -> Set<Element> {
         return Set(self)
     }
 }
 
-extension Array {
+public extension Array {
+    //TODO: Evaluate switching from ip_safe to safe here
```

Comment:
I think we can remove this todo-since its unlikely we are going to not namespace extensions.